src/main/scala/ch/epfl/bluebrain/nexus/kg/routes/ViewRoutes.scala

```diff
@@ -208,58 +208,93 @@ class ViewRoutes private[routes] (
   }
 
   private def sparqlRoute: Route =
-    (post & pathPrefix(IdSegment / "sparql") & pathEndOrSingleSlash) { id =>
-      operationName(s"/${config.http.prefix}/views/{}/{}/{}/sparql") {
-        hasPermission(query).apply {
-          entity(as[String]) { query =>
-            val result: Task[Either[Rejection, SparqlResults]] = viewCache.getBy[View](project.ref, id).flatMap {
-              case Some(v: SparqlView) => sparqlQuery(v, query).map(Right.apply)
-              case Some(agg: AggregateSparqlView[_]) =>
-                val resultListF = agg.queryableViews.flatMap { views =>
-                  Task.gatherUnordered(views.map(sparqlQuery(_, query)))
-                }
-                resultListF.map(list => Right(list.foldLeft(SparqlResults.empty)(_ ++ _)))
-              case _ => Task.pure(Left(NotFound(id.ref)))
+    concat(
+      (post & pathPrefix(IdSegment / "sparql") & pathEndOrSingleSlash) { id =>
+        operationName(s"/${config.http.prefix}/views/{}/{}/{}/sparql") {
+          hasPermission(query).apply {
+            entity(as[String]) { query =>
+              val result = viewCache.getBy[View](project.ref, id).flatMap(runSearch(id, query))
+              complete(result.runWithStatus(StatusCodes.OK))
+            }
+          }
+        }
+      },
+      (post & pathPrefix(IdSegment / IdSegment / "sparql") & pathEndOrSingleSlash) { (id, projId) =>
+        operationName(s"/${config.http.prefix}/views/{}/{}/{}/{}/sparql") {
```

Comment:
how about changing the APIs to this:
`/{prefix}/views/{org}/{proj}/{view_id}/projections/{projection_id}/*`